rclpy/src/rclpy/_rclpy.c

```diff
@@ -851,6 +852,99 @@ rclpy_count_subscribers(PyObject * Py_UNUSED(self), PyObject * args)
   return _count_subscribers_publishers(args, "subscribers", rcl_count_subscribers);
 }
 
+typedef rcl_ret_t (* rcl_get_info_by_topic_func_t)(
+  const rcl_node_t * node,
+  rcutils_allocator_t * allocator,
+  const char * topic_name,
+  bool no_mangle,
+  rmw_topic_info_array_t * info_array);
+
+static PyObject *
+_get_info_by_topic(
+  PyObject * args, const char * type,
```

Comment:
@jaisontj @mm318 is `type` unused?

Reply:
`type` is a string representing "publishers" or "subscriptions". It is unused in that it is only used for logging to distinguish between `rcl_get_publishers_info_by_topic` and `rcl_get_subscribers_info_by_topic`